### ISSUE TITLE
charset=utf-8 for application/x-www-form-urlencoded causes HTTP response 415

### ISSUE THREAD

**Issue by dreamerlzl** [poem-web/poem#395]
## question
when I am using HTTPie to test an endpoint with `http -f POST http://localhost:8081/subscription name=foo`, the following error is reported

```
HTTP/1.1 415 Unsupported Media Type
content-length: 116
date: Wed, 21 Sep 2022 03:19:31 GMT

invalid content type `application/x-www-form-urlencoded; charset=utf-8`, expect: `application/x-www-form-urlencoded`
```
(httpie will automatically add `charset=utf-8` in a form POST request)
Is this an expected behavior?

## version
```yaml
poem = { version = "1.3.40", features = ["test"] }
poem-openapi = { version = "2.0.11", features = ["swagger-ui"] }
```

## my test code
my source code is here: https://github.com/dreamerlzl/zero2prod/blob/main/src/routes/subscribe.rs

**Comment by sunli829** [comment]
I haven't figured out a solution to this problem, so it can only be `application/x-www-form-urlencoded`. 😂

**Comment by banool** [comment]
I'm getting this error too, even without the `charset=utf8` part, and unfortunately I don't have control over my clients here. Is there some way to work around this?
```
the `Content-Type` requested by the client is not supported: application/x-www-form-urlencoded
```

**Comment by banool** [comment]
Okay I updated Poem and the error randomly disappeared 🤔. I can't see a change that would fix this but I'll accept it lol.

**Comment by brunobell** [comment]
@sunli829 Do you think it's appropriate to update poem's code to use `mime`'s [`essence_str`](https://docs.rs/mime/0.3.16/mime/struct.Mime.html#method.essence_str) to decide the `content-type` of the request? Thus the `charset` parameter should not be a problem.
If it's ok, I'd be happy to shoot a PR.

**Comment by sunli829** [comment]
I've fixed this, please help me test it. 🙂

**Comment by dreamerlzl** [comment]
> I've fixed this, please help me test it. slightly_smiling_face

Thank you! But after setting
```yaml
poem-openapi = { git = "https://github.com/poem-web/poem", rev="a43e94f2cfbefdbb484e3bdb51fafdb05201dbbb", features = ["swagger-ui"] }
```
my project doesn't compile, saying that `the trait `poem::Endpoint` is not implemented for `impl poem::endpoint::endpoint::Endpoint``. I would fix that and try again soon.

**Comment by sunli829** [comment]
Released in `v1.3.47`
